docs/source/basics/prompt_templates.rst

```diff
@@ -0,0 +1,101 @@
+.. _prompt_templates_usage_label:
+
+================
+Prompt Templates
+================
+
+Prompt templates are structured text templates that user prompts are formatted with
+for optimal model performance on a task. They can serve many purposes:
+
+1. Model-specific templates that are required whenever the model is prompted, such as the [INST]
+   tags in Llama2 and in Mistral. These models were pre-trained with these tags and using them
+   in inference can help ensure optimal performance.
+2. Task-specific templates to gear models for a particular task that it will expect after training.
+   Example include grammar correction (:class:`~torchtune.data.GrammarErrorCorrectionTemplate`),
+   summarization (:class:`~torchtune.data.SummarizeTemplate`), question answering (:class:`~torchtune.data.QuestionAnswerTemplate`),
+   and more.
+3. Community standardized templates, such as :class:`~torchtune.data.ChatMLTemplate`
+
+The added text is different from special tokens that are added by the model tokenizer. For an extended
+discussion on the different between prompt templates and special tokens, see :ref:`prompt_template_vs_special_tokens`.
+
+Custom prompt templates
+-----------------------
+In most cases, you can define your prompt template as a dictionary and pass it directly into the tokenizer.
+The expected format is:
+
+.. code-block:: python
+
+    template = {role: (PREPEND_TAG, APPEND_TAG)}
```

Comment:
Could you give a concrete example would look like? e.g. with this template, prompts will be formatted like so...? might be overkill if we're already showing above but significantly reduces cognitive load

Reply:
I actually see these points are covered in the tokenizer page where there are more concrete examples, but that page defers here for further details. I still think some degree of self-containedness is good here, particularly if we reference this page in other parts of the docs